lib/features/extend_task_graph.js

```diff
@@ -114,13 +122,13 @@ ExtendTaskGraph.prototype = {
     }
 
     // Iterate through the graphs extending where possible.
-    var extensions = payload.graphs.map(function(graph) {
-      return this.extendTaskGraph(taskHandler, graph);
-    }, this);
+    await Promise.all(payload.graphs.map((graph) => {
```

Comment:
async returns a promise so your map will now return an array of promises which `Promise.all` can deal with.